pkg/controller/kieapp/defaults/defaults_test.go

```diff
@@ -7491,3 +7497,179 @@ func TestRhpamKieserverEnvWithoutDecisionsOnlyEnabled(t *testing.T) {
 	result, _ := strconv.ParseBool(getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_DECISIONS_ONLY"))
 	assert.False(t, result)
 }
+
+func TestCredentialWithCredentialSecretAndCredentialsUseAsDefault(t *testing.T) {
+	// with all credentials provided the behaviour use the legacy with username and password
+	username := "admin"
+	password := "password"
+	myNamespace := "myNamespace"
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: myNamespace,
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhpamProduction,
+			CommonConfig: api.CommonConfig{
+				AdminPassword:          password,
+				AdminUser:              username,
+				SecretAdminCredentials: constants.KIE_ADMIN_CREDENTIALS_SECRET,
+			},
+		},
+	}
+
+	service := test.MockService()
+	env, err := GetEnvironment(cr, service)
+	assert.Nil(t, err, "Error getting prod environment")
+	providedSecret := corev1.Secret{}
+	errS := service.Get(context.TODO(), types.NamespacedName{
+		Name:      constants.KIE_ADMIN_CREDENTIALS_SECRET,
+		Namespace: myNamespace,
+	}, &providedSecret)
+	assert.Nil(t, errS, "Error secret provided")
+	adminUSer := providedSecret.StringData[constants.USERNAME_ADMIN_SECRET_KEY]
+	adminPassword := providedSecret.StringData[constants.PASSWORD_ADMIN_SECRET_KEY]
+	assert.Equal(t, adminUSer, username)
+	assert.Equal(t, adminPassword, password)
+	adminUserEnv := getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_ADMIN_USER")
+	adminPasswordEnv := getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_ADMIN_PWD")
+	assert.Equal(t, len(adminUserEnv), 0)
+	assert.Equal(t, len(adminPasswordEnv), 0)
+}
+
+func TestCredentialWithCredentialSecretAndCredentials(t *testing.T) {
```

Comment:
maybe this tests name fits better, wdyt?
```suggestion
func TestCredentialWithCredentialSecretAndDefaultUsernameAndPassword(t *testing.T) {
```

Reply:
Changed with different name